container.go

```diff
@@ -726,6 +726,14 @@ func (c *Client) Stats(opts StatsOptions) (retErr error) {
 		close(errC)
 	}()
 
+	go func() {
+		// block here waiting for the signal to stop function
+		select {
+		case <-opts.Done:
+			readCloser.Close()
+		}
+	}()
+
```

Comment:
I'd like to be able to prevent this goroutine from leaking. I.e.: if done never gets closed or a value, it will block forever. You could create another channel, and add another case to the select statement, something like:

```go
quit := make(chan struct{})
defer close(quit)
go func() {
	select {
	case <-opts.Done:
		readCloser.Close()
	case <-quit:
		return
	}
}
```

Does it make sense?